### PR TITLE
Add unit tests for comma split kafka adapter

### PR DESCRIPTION
**Pull request by lberk**
Unit tests to match #1065

cc/ @duglin

### CODE REVIEW

File: kafka/source/pkg/adapter/adapter_test.go [hunk 1]
```diff
@@ -628,3 +629,48 @@ func TestAdapterStartFailure(t *testing.T) {
...
+			[]string{
+				"url-one.kafka:9092",
+				"url-two.kafka:9092",
+				"url-three.kafka:9092",
+			},
+		},
+	} {
+		t.Run(tt.name, func(t *testing.T) {
+			got := SplitOnCommas(tt.bootstrapServers)
+			if diff := cmp.Diff(tt.want, got); diff != "" {
```
Comment (duglin) [inline kafka/source/pkg/adapter/adapter_test.go]: just wondering why you used `cmp` instead of `reflect.DeepEqual` ?

**Comment by duglin** [comment]
thanks for doing this. The only other test I would add is for `,,one,two,,three,four,,`.
Because your repeated commas case (`,,`) could work due to leading or trailing duplicate `,` checking, and not necessarily for middle `,` dup checking. This one case above should get all 3 cases in one shot.

File: kafka/source/pkg/adapter/adapter_test.go (hunk 1 above)
Comment (lberk) [reply kafka/source/pkg/adapter/adapter_test.go]: Path of least surprise in the codebase
```
% git grep -i cmp.diff | grep -v ^vendor | grep _test.go | wc -l
63
% git grep -i reflect.deep | grep -v ^vendor | grep _test.go | wc -l
0

**Comment by lberk** [comment]
@duglin added the extra testcase.

File: kafka/source/pkg/adapter/adapter_test.go (hunk 1 above)
Comment (duglin) [reply kafka/source/pkg/adapter/adapter_test.go]: ah didn't realize we were using it in other spots - thanks for the info

**Comment by duglin** [comment]
LGTM

**Pull request merged**